spirv_msl.cpp

```diff
@@ -96,6 +100,25 @@ string CompilerMSL::compile()
 	return compile(default_msl_cfg, nullptr, nullptr);
 }
 
+// Register the need to output any custom functions.
+void CompilerMSL::register_custom_functions()
```

Comment:
traverse_reachable_opcodes would be a bit better here as here we'd emit mod() even if a function isn't actually used.